File: 2140_solving_questions_with_brainpower.cpp
class Solution {
public:
    long long mostPoints(vector<vector<int>>& k) {
        int l = k.size();
        vector<long long> m(l + 1);
        for (int n = l - 1; n >= 0; --n)
            m[n] = max(k[n][0] + ((n + k[n][1] + 1 < l) ? m[n + k[n][1] + 1] : 0), m[n + 1]);
        return m[0];
    }
};

File: 2140_solving_questions_with_brainpower.java
class Solution {
    public long mostPoints(int[][] e) {
        int f = e.length;
        long[] g = new long[f + 1];
        for (int h = f - 1; h >= 0; --h)
            g[h] = Math.max(e[h][0] + ((h + e[h][1] + 1 < f) ? g[h + e[h][1] + 1] : 0), g[h + 1]);
        return g[0];
    }
}
